protocol/messenger.go

```diff
@@ -5723,3 +5897,79 @@ func (m *Messenger) handleSyncClearHistory(state *ReceivedMessageState, message
 	})
 	return nil
 }
+
+func (m *Messenger) handleSyncTrustedUser(state *ReceivedMessageState, message protobuf.SyncTrustedUser) error {
+	updated, err := m.verificationDatabase.UpsertTrustStatus(message.Id, verification.TrustStatus(message.Status), message.Clock)
+	if err != nil {
+		return err
+	}
+
+	if updated {
+		state.AllTrustStatus[message.Id] = verification.TrustStatus(message.Status)
+
+		contact, ok := m.allContacts.Load(message.Id)
+		if !ok {
+			m.logger.Info("contact not found")
+			return nil
+		}
+
+		contact.TrustStatus = verification.TrustStatus(message.Status)
+		m.allContacts.Store(contact.ID, contact)
+		state.ModifiedContacts.Store(contact.ID, true)
+	}
+
+	return nil
+}
+
+func ToVerificationRequest(message protobuf.SyncVerificationRequest) *verification.Request {
```

Comment:
Would be cool if this went into `protocol/requests/...` instead for consistency's sake, but doesn't have to